slowquant/qiskit_interface/util.py

```diff
@@ -68,12 +68,14 @@ def get_bitstring_sign(op: Pauli, binary: str) -> int:
     Returns:
         Expectation value of Pauli string.
     """
-    sign = 1
+    count = 0
     for i, pauli in enumerate(op.to_label()):
         if not pauli == "I":
             if binary[i] == "1":
-                sign = sign * (-1)
-    return sign
+                count += 1
+    if count % 2 == 1:
+        return -1
+    return 1
```

Comment:
Is there a reason for this re-write?

Reply:
The line `sign = sign * (-1)` was a bottleneck.
This is much faster